resolver/src/p2p/engine.rs

```diff
@@ -170,10 +180,18 @@ impl<
             self.metrics.serve_processing.set(self.serves.len() as i64);
 
             // Update peer list if-and-only-if it might have changed.
-            let peer_set_id = self.coordinator.peer_set_id();
-            if self.last_peer_set_id != Some(peer_set_id) {
-                self.last_peer_set_id = Some(peer_set_id);
-                self.fetcher.reconcile(self.coordinator.peers());
+            let peer_set_id = self.peer_provider.latest_peer_set().await;
```

Comment:
It still feels wrong to hammer this so hard (blocking channel request per loop).